### PR TITLE
[docs] Talk about running a server in tutorial

### PR DESCRIPTION
**Pull request by zpao**
We've talked about this a few times and even half done it. This is for real.

I also changed the name of the html file to be `index.html` (that's what's in the tutorial repo)

See also #2004, #2739, #800 (eh, not totally but I want to close it), https://github.com/reactjs/react-tutorial/issues/1 and surely some others.

### CODE REVIEW

File: docs/docs/tutorial.md [hunk 1]
```diff
@@ -379,7 +385,7 @@ When the component is first created, we want to GET some JSON from the server an
 
 We'll use jQuery to help make an asynchronous request to the server.
 
-Note: because this is becoming an AJAX application you'll need to develop your app using a web server rather than as a file sitting on your file system. The easiest way to do this is to run `python -m SimpleHTTPServer` in your application's directory.
+Note: because this is becoming an AJAX application you'll need to develop your app using a web server rather than as a file sitting on your file system. [As mentioned above](#running-a-server), we have provided serveral servers you can use [on GitHub](https://github.com/reactjs/react-tutorial/). The provide the functionality you need for the rest of this tutorial.
```
Comment (gaearon) [inline docs/docs/tutorial.md]: "The provide" typo

Comment (zpao) [inline docs/docs/tutorial.md]: Nice catch, fixed.

**Comment by zpao** [comment]
Anybody have objections? @spicyj?

**Comment by spicyj** [comment]
Seems reasonable. Did you test that loading the first parts from a file: URL works properly in Firefox and Chrome and Safari? I imagine it would since we're not doing anything fancy but it can't hurt to check.

**Comment by brigand** [comment]
Awesome, there's been a few stackoverflow questions about this.

But I'd much rather see '/api/comments' than 'comments.json', which indicates a static file.

**Comment by zpao** [comment]
Yea, I guess moving it to `/api/comments` would be ok. I don't feel strongly, as long as its understandable and works (we needed `comments.json` before because of the simple server assumption).

Per the $.ajax bit, meh. I think it's ok and reduces the number of things that needed to be learned right away. It might be good to have an "advanced" tutorial which goes back through and talks about more topics like mixins, data stores. Overall we probably need a best practices section and should talk about architecture a bit but I don't know the right place.

**Pull request merged**